packages/web/components/complex/asset-variants-conversion/asset-variants-conversion-modal.tsx

```diff
@@ -0,0 +1,431 @@
+        })) ?? [],
+      [allocationData]
+    );
+
+    console.log("prepareRouteInputs", prepareRouteInputs);
+
+    const apiUtils = api.useUtils();
+
+    const quotes = useQueries({
+      queries: prepareRouteInputs.map(
+        ({ tokenInDenom, tokenInAmount, tokenOutDenom }) => ({
+          queryKey: [
+            "routeTokenOutGivenIn",
+            tokenInDenom,
+            tokenInAmount,
+            tokenOutDenom,
+          ],
+          queryFn: () =>
+            apiUtils.local.quoteRouter.routeTokenOutGivenIn.fetch(
+              {
+                tokenInDenom,
+                tokenInAmount,
+                tokenOutDenom,
+              },
+              {}
+            ),
+          staleTime: Infinity,
+        })
+      ),
+    });
+
+    const isLoadingQuotes = quotes.some((result) => result.isLoading);
+
+    const dataQuotes = quotes.map((result) => result.data);
+
+    // console.log("dataQuotes:", dataQuotes);
+
+    // const { data: routeData, isLoading: isRouteLoading } =
+    //   api.local.quoteRouter.routeTokensOutGivenIn.useQuery(prepareRouteInputs, {
+    //     enabled: !!allocationData,
+    //     refetchOnWindowFocus: false,
+    //   });
+
+    const convertSelectedAssets = async () => {
+      const allSwapMessages = [];
+
+      for (const quote of dataQuotes) {
+        console.log("quote", quote);
+        if (!quote) continue;
+
+        // add slippage to out amount from quote
+        let slippage = new Dec(0.05);
+        // if it's an alloy, or CW pool, let's assume it's a 1:1 swap
+        // so, let's remove slippage to convert more of the asset
+        if (
+          quote.split.length === 1 &&
+          quote.split[0].pools.length === 0 &&
+          quote.split[0].pools[0].type.startsWith("cosmwasm")
+        ) {
+          slippage = new Dec(0);
+        }
+
+        const swapMessagesConfig = {
+          quote: quote,
+          // TODO - refactor this to be cleaner
+          tokenOutCoinMinimalDenom: quote.amount.currency.coinMinimalDenom,
+          tokenOutCoinDecimals: quote.amount.currency.coinDecimals,
+          tokenInCoinDecimals:
+            checkedVariants.find(
+              (variant) =>
+                variant.canonicalAsset?.coinMinimalDenom ===
+                quote.amount.currency.coinMinimalDenom
+            )?.asset?.coinDecimals ?? 0,
+          tokenInCoinMinimalDenom:
+            checkedVariants.find(
+              (variant) =>
+                variant.canonicalAsset?.coinMinimalDenom ===
+                quote.amount.currency.coinMinimalDenom
+            )?.asset?.coinMinimalDenom ?? "",
+          maxSlippage: slippage.toString(),
+          // coinAmount: quote.amount.toCoin().amount,
+          coinAmount: quote.split[0].initialAmount.toString(),
+          userOsmoAddress: account?.address ?? "",
+          quoteType: "out-given-in" as QuoteType,
+        };
+
+        const swapMessages = await getSwapMessages(swapMessagesConfig);
```

Comment:
It would be more efficient to use Promise.all

Remove the await from `getSwapMessages` and push them to the array as promises. Then call `Promise.all(allSwapMessages)`